### ISSUE TITLE
Config hot reloading

### ISSUE THREAD

**Issue by Cryrivers** [Cryrivers/manta-style#6]

**Comment by tanhauhau** [comment]
potential challenges i can see here is that, not just config file changes to be watched, all the dependencies of the config file change will affect the type exported by the config file too.

**Comment by Cryrivers** [comment]
Yep. Should be easier for TS compiler to do as I can get a list of dependencies. And TS compiler has incremental compile built-in.

Not sure for Babel and GraphQL tho.

**Comment by JennieJi** [comment]
<img width="738" alt="screen shot 2018-08-19 at 10 29 05 pm" src="https://user-images.githubusercontent.com/1913045/44309882-d6c4a680-a3ff-11e8-8475-57cf573a8e13.png">
Guys, I've supported config file change hot reload, but I haven't check dependency in my changes. Seems quite challenging...
Well, let's support config file itself first. Pls check.

**Comment by Cryrivers** [comment]
Orz. If you do a full rebuild then no need to take account of dependency issue.

**Comment by JennieJi** [comment]
@Cryrivers Not really... I mean I do use @manta-style/typescript-builder to rebuild the whole config file.   
But to know whether dependencies are updated, you have to watch on them. To watch on them, you have to find out all the dependent files by something like babel. I think that's what @tanhauhau means.  
For this, I think it is not that necessary to do this complicated work.

**Comment by Cryrivers** [comment]
Oh, now I get what you meant.

For TypeScript builder, you might want to check your screenshot. TypeScript Compiler API returns an object containing `emittedFiles` property, which contains a list of file path that can be considered as dependencies.

(however, typescript builder only returns the dest path of entrypoint instead of full list of emitted files. 😂)

**Comment by JennieJi** [comment]
Yes, I saw it.   
My suggestion is watching the config file only, and once it changed, let type script builder to handle and build a brand new file.  
I realize that there is no meaning to watch dependent files at all, because config file might change it's
